test/unit/SyncLog/Storage/RdsSyncLogStorageTest.php

```diff
@@ -0,0 +1,130 @@
+use oat\taoSync\model\SyncLog\SyncLogEntity;
+
+/**
+ * Class RdsSyncLogStorageTest
+ */
+class RdsSyncLogStorageTest extends TestCase
+{
+    const PERSISTENCE_ID = 'PERSISTENCE_ID';
+
+    /**
+     * @var RdsSyncLogStorage
+     */
+    private $object;
+
+    /**
+     * @var \common_persistence_Manager|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $persistenceManagerMock;
+
+    /**
+     * @var \common_persistence_SqlPersistence|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $persistenceMock;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->persistenceMock = $this->createMock(\common_persistence_SqlPersistence::class);
+
+        $this->persistenceManagerMock = $this->createMock(\common_persistence_Manager::class);
+        $this->persistenceManagerMock ->method('getPersistenceById')
+            ->willReturn($this->persistenceMock);
+
+        $serviceLocatorMock = $this->getServiceLocatorMock([
+            \common_persistence_Manager::SERVICE_ID => $this->persistenceManagerMock
+        ]);
+
+        $this->object = new RdsSyncLogStorage([
+            RdsSyncLogStorage::OPTION_PERSISTENCE_ID => self::PERSISTENCE_ID
+        ]);
+        $this->object->setServiceLocator($serviceLocatorMock);
+    }
+
+    /**
+     * Test that constructor throws exception when "persistenceId" option is missing.
+     *
+     * @throws MissingOptionException
+     */
+    public function testConstructWithoutRequiredOption()
+    {
+        $this->expectException(MissingOptionException::class);
+
+        new RdsSyncLogStorage([]);
+    }
+
+    /**
+     * Test getPersistence method.
+     */
+    public function testGetPersistence()
+    {
+        $persistence = $this->object->getPersistence();
+
+        $this->assertInstanceOf(\common_persistence_SqlPersistence::class, $persistence, 'Returned persistence must be an object of expected type.');
+    }
+
+    /**
+     * Test create method.
+     */
+    public function testCreate()
+    {
+        $tableName = 'synchronisation_log';
+        $syncId = 'SYNC_ID';
+        $boxId = 'BOX_ID';
+        $orgId = 'ORG_ID';
+        $data = ['ENTITY_TYPE' => ['ACTION' => 'AMOUNT']];
+        $status = 'SYNC_STATUS';
+        $createdAt = new \DateTime('2019-01-01 12:00:00');
+        $report = $this->createMock(\common_report_Report::class);
+        $report->expects($this->once())
+            ->method('JsonSerialize')
+            ->willReturn(['data' => 'REPORT_DATA']);
+
+        $expectedInsertData = [
+            'sync_id' => $syncId,
+            'box_id' => $boxId,
+            'organization_id' => $orgId,
+            'data' => '{"ENTITY_TYPE":{"ACTION":"AMOUNT"}}',
+            'status' => $status,
+            'report' => '{"data":"REPORT_DATA"}',
+            'created_at' => '2019-01-01 12:00:00',
+        ];
+
+        $this->persistenceMock->expects($this->once())
```

Comment:
Unfortunately this is not proper test. 
Please test all the public methods and use real sql storage in order to test that data was successfully inserted/updated.

Reply:
Fixed